src/fondant/core/component_spec.py

```diff
@@ -464,25 +464,34 @@ def _operation_schema_to_dataset_schema(self, name: str) -> t.Mapping[str, Field
         Args:
             name: "consumes" or "produces"
         """
-        spec_mapping = getattr(self, f"operations_{name}")
-        args_mapping = self._mappings[name]
+        operations_schema = getattr(self, f"operations_{name}")
 
-        if not args_mapping:
-            return spec_mapping
+        # Dataset arguments mapping
+        column_mapping_args = self._mappings[name]
 
-        mapping = dict(spec_mapping)
+        if not column_mapping_args:
+            return operations_schema
 
-        for key, value in args_mapping.items():
-            if not isinstance(value, str):
+        mapping = dict(operations_schema)
+
+        for (
+            operations_column_name,
+            dataset_column_name_or_type,
+        ) in column_mapping_args.items():
+            # If the value is not a string, it means it's a type, so we skip it
+            if not isinstance(dataset_column_name_or_type, str):
                 continue
 
-            if key in spec_mapping:
-                mapping[value] = Field(name=value, type=mapping.pop(key).type)
+            if operations_column_name in operations_schema:
```

Comment:
We are able to add a check for `addtionalProperties` here, and ignore this exception if `additionalProperties` are allowed. However, in that case we have to change the string mapping to a `string: pa.DataType` mapping to avoid issues later on. The `ComponentOp` would need to access the dataset schema.
I think it might be cleaner to implement this behaviour before we are initialising the `ComponentOp`, infer the consumes, validate and fix it (if needed). The `ComponentOp` will be initialised with a valid consumes definition afterwards.

Reply:
Agree, see my comment below.